components/inputgroup/index.css

```diff
@@ -10,11 +10,453 @@ OF ANY KIND, either express or implied. See the License for the specific languag
+    }
+
+    &:focus, &:focus-visible, &.is-focused, &.is-keyboardFocused {
+      border: none;
+
+      &::after {
+        box-shadow: none;
+      }
+    }
+  }
+
+  .spectrum-InputGroup-button:not(:disabled):not(.is-invalid) {
+    .spectrum-PickerButton-fill {
+      /* Subcomponent token override */
+      --spectrum-alias-infieldbutton-fill-border-color-default: var(--spectrum-combobox-border-color-default);
+      border-color: var(--mod-combobox-border-color-default, var(--spectrum-alias-infieldbutton-fill-border-color-default));
+    }
+
+    &:focus, &.is-focused, &.is-keyboardFocused {
+      .spectrum-PickerButton-fill:not(.is-invalid) {
+        border-color: var(--mod-combobox--border-color-key-focus, var(--spectrum-alias-infieldbutton-fill-border-color-key-focus));
+      }
+    }
+  }
+
+  & .is-open.spectrum-Popover {
+    transform: translateY(var(--mod-combobox-spacing-edge-to-menu, var(--spectrum-combobox-spacing-edge-to-menu)));
+  }
+}
+
+.spectrum-InputGroup.is-invalid {
+  .spectrum-InputGroup-input {
+
+    border-color: var(--mod-combobox-border-color-invalid-default, var(--spectrum-combobox-border-color-invalid-default));
+
+    &:hover {
+      border-color: var(--mod-combobox-border-color-invalid-hover, var(--spectrum-combobox-border-color-invalid-hover));
+    }
+
+    &:active {
+      border-color: var(--mod-combobox-border-color-invalid-active, var(--spectrum-combobox-border-color-invalid-active));
+    }
+
+    &.focus-ring {
+      box-shadow: none;
+      /* Subcomponent token override */
+      --spectrum-alias-infieldbutton-fill-border-color-key-focus: var(--spectrum-combobox-border-color-invalid-key-focus);
+      border-color: var(--mod-combobox-border-color-key-focus, var(--spectrum-alias-infieldbutton-fill-border-color-key-focus));
+    }
+  }
+
+  .spectrum-Textfield-validationIcon {
+    color: var(--mod-combobox-alert-icon-color, var(--spectrum-combobox-alert-icon-color));
+  }
+}
+
+.spectrum-InputGroup.is-disabled {
+  .spectrum-InputGroup-input {
+    background-color: var(--mod-combobox-background-color-disabled, var(--spectrum-combobox-background-color-disabled));
+
+    &::placeholder {
+      color: var(--mod-combobox-font-color-disabled, var(--spectrum-combobox-font-color-disabled));
+    }
+  }
+}
+
+.spectrum-InputGroup-progress-circle {
+  position: absolute;
+  inset-inline-end: calc(
+    var(--mod-combobox-spacing-inline-icon-to-button, var(--spectrum-combobox-spacing-inline-icon-to-button)) +
+    var(--mod-combobox-button-size, var(--spectrum-combobox-button-size))
+  );
+  inset-block-start: var(--mod-combobox-block-spacing-edge-to-progress-circle, var(--spectrum-combobox-block-spacing-edge-to-progress-circle));
+  inset-block-end: var(--mod-combobox-block-spacing-edge-to-alert, var(--spectrum-combobox-block-spacing-edge-to-alert));
+}
+
+/* Overrides inset-inline position from spectrum-Textfield.is-invalid */
```

Comment:
This overrides the styles within the `.spectrum-Textfield` subcomponent of Combobox. I chose to override the selector rather than the tokens in this case because there is more than one token required to derive the `inset-inline-end` value.